crates/relayer/src/chain/cosmos.rs

```diff
@@ -318,6 +319,68 @@ impl CosmosSdkChain {
         Ok(params)
     }
 
+    /// Query the node for its configuration parameters.
+    ///
+    /// ### Note: This query endpoint was introduced in SDK v0.46.3/v0.45.10. Not available before that.
+    ///
+    /// Returns:
+    ///     - `Ok(Some(..))` if the query was successful.
+    ///     - `Ok(None) in case the query endpoint is not available.
+    ///     - `Err` for any other error.
+    pub fn query_config_params(&self) -> Result<Option<ConfigResponse>, Error> {
+        crate::time!("query_config_params");
+        crate::telemetry!(query, self.id(), "query_config_params");
+
+        // Helper function to diagnose if the node config query is unimplemented
+        // by matching on the error details.
+        fn is_unimplemented_node_query(err_status: tonic::Status) -> bool {
+            if err_status.code() != tonic::Code::Unimplemented {
+                return false;
+            }
+
+            err_status
+                .message()
+                .contains("unknown service cosmos.base.node.v1beta1.Service")
+        }
+
+        let mut client = self
+            .block_on(
+                ibc_proto::cosmos::base::node::v1beta1::service_client::ServiceClient::connect(
+                    self.grpc_addr.clone(),
+                ),
+            )
+            .map_err(Error::grpc_transport)?;
+
+        let request = tonic::Request::new(ibc_proto::cosmos::base::node::v1beta1::ConfigRequest {});
+
+        match self.block_on(client.config(request)) {
+            Ok(response) => {
+                let params = response.into_inner();
+
+                Ok(Some(params))
+            }
+            Err(e) => {
+                if is_unimplemented_node_query(e.clone()) {
```

Comment:
so that we don't need to clone it here

```suggestion
                if is_unimplemented_node_query(&e) {
```